daiquiri/datalink/adapter.py

```diff
@@ -0,0 +1,165 @@
+               'service_def': row[3] or '',
+               'error_message': row[4] or '',
+               'description': row[5] or '',
+               'semantics': row[6],
+               'content_type': row[7] or '',
+               'content_length': row[8] or 0
+            }
+        ]
+
+
+class MetadataDatalinkAdapterMixin(object):
+
+    def get_schema_list(self):
+        from daiquiri.metadata.models import Schema
+
+        for schema in Schema.objects.iterator():
+            yield 'schema', schema
+
+    def get_schema_identifier(self, schema):
+        return schema.name
+
+    def get_schema_links(self, schema):
+        schema_links = []
+
+        if schema.metadata_access_level == ACCESS_LEVEL_PUBLIC:
+            identifier = self.get_schema_identifier(schema)
+
+            path = reverse('metadata:schema', args=[schema.name])
+            access_url = settings.SITE_URL.rstrip('/') + path
+
+            schema_links = [{
+               'ID': identifier,
+               'access_url': access_url,
+               'service_def': '',
+               'error_message': '',
+               'description': 'Documentation for the {} schema'.format(schema.name),
+               'semantics': '#documentation',
+               'content_type': 'application/html',
+               'content_length': None
+            }]
+
+            if schema.doi:
+                schema_links.append({
+                   'ID': identifier,
+                   'access_url': get_doi_url(schema.doi),
+                   'service_def': '',
+                   'error_message': '',
+                   'description': 'Digital object identifier',
+                   'semantics': '#doi',
+                   'content_type': 'application/html',
+                   'content_length': None
+                })
+
+        return schema_links
+
+    def get_table_list(self):
+        from daiquiri.metadata.models import Table
+
+        for table in Table.objects.iterator():
+            yield 'table', table
+
+    def get_table_identifier(self, table):
+        return '{}.{}'.format(table.schema.name, table.name)
+
+    def get_table_links(self, table):
+        table_links = []
+
+        if table.metadata_access_level == ACCESS_LEVEL_PUBLIC and \
+                table.schema.metadata_access_level == ACCESS_LEVEL_PUBLIC:
+            identifier = self.get_table_identifier(table)
+
+            path = reverse('metadata:table', args=[table.schema.name, table.name])
+            access_url = settings.SITE_URL.rstrip('/') + path
+
+            table_links.append({
+               'ID': identifier,
+               'access_url': access_url,
+               'service_def': '',
+               'error_message': '',
+               'description': 'Documentation for the {} table'.format(table.name),
+               'semantics': '#documentation',
+               'content_type': 'application/html',
+               'content_length': None
+            })
+
```

Comment:
I would add the `related_identifiers` here as well. 

However they can have so many shapes: there are quite a few `related_identifiers_type` and `relation_type` and they do not directly translate to `datalink`'s `description` and `semantic`. 

We would first need to come up with a translation map from `related_identifiers` `type` and `relation_type` to  `datalink` `semantic` and `description`, I guess.

Reply:
Done, I created descriptions with some sting handling here: https://github.com/django-daiquiri/daiquiri/blob/datalink/daiquiri/datalink/constants.py#L45. I don't think we need to care about `related_identifiers_type` here.